### ISSUE TITLE
Wrong result when parsing CJK(中文/日文/韩文) char followed by colon

### ISSUE THREAD

**Issue by mercury233** [gregjacobs/Autolinker.js#394]
```javascript
const Autolinker = require("autolinker")
const autolinker = new Autolinker({
	urls: {
		schemeMatches: true,
		tldMatches: false,
		ipV4Matches: false
	},
	newWindow: false,
	stripPrefix: false,
	email: false,
	phone: false,
	stripTrailingSlash: false
});

console.log(autolinker.link("链接：https://www.google.com"));
console.log(autolinker.link("link:https://www.google.com"));
console.log(autolinker.link("链接: https://www.google.com"));
console.log(autolinker.link("链接:https://www.google.com"));
```
![image](https://user-images.githubusercontent.com/13391795/200872382-fd93f67a-b2e7-465d-8a43-2188922005f4.png)

https://runkit.com/embed/xpmv085le79y

The 4th result is wrong

**Comment by keisukekomeda** [comment]
Similar example here.

```TypeScript
console.log(Autolinker.link("こちら→https://google.com"))
// result: こちら→https://g<a href="http://oogle.com" target="_blank" rel="noopener noreferrer">oogle.com</a>
```

**Comment by gregjacobs** [comment]
This is now fixed in 4.0.1. Thanks for reporting!

**Comment by gregjacobs** [comment]
This is now fixed in 4.0.1. Thanks for reporting!
